### PR TITLE
Try to make points() a little less heavyweight

### PR DESCRIPTION
**Pull request by agcapps**
This starts to remove the type insurance code since we are fairly sure we will get index_t.  Needs testing.

### CODE REVIEW

**Comment by agcapps** [comment]
@cyrush , @aaroncblack and I talked about this routine.  It seems this takes most of the time in a run he is interested in, so this was a first pass at speeding things up.  Please let me know if this is starting to go down the right path.

**Comment by cyrush** [comment]
This should certainly help.
This is one of those areas that is using `std::set`  as well.
I assume the algorithm relies on that to de-duplicate.
Do the set operations show up in profiles?

**Comment by cyrush** [comment]
Seems like there is an indexing issue. 


```
unknown file : error : C++ exception with description " [D:\a\conduit\conduit\build\RUN_TESTS.vcxproj]
  file: D:\a\conduit\conduit\src\libs\conduit\conduit_node.cpp
  line: 14021
  message: 
  Invalid child index: 0 (number of children: 0)
  " thrown in the test body.
unknown file : error : C++ exception with description " [D:\a\conduit\conduit\build\RUN_TESTS.vcxproj]
  file: D:\a\conduit\conduit\src\libs\conduit\conduit_node.cpp
  line: 14021
  message: 
  Invalid child index: 0 (number of children: 0)
  " thrown in the test body.
  [  FAILED  ] conduit_blueprint_mpi_mesh_transform.generate_sides (2 ms)
  [  FAILED  ] conduit_blueprint_mpi_mesh_transform.generate_sides (2 ms)
  [ RUN      ] conduit_blueprint_mpi_mesh_transform.generate_corners
  [ RUN      ] conduit_blueprint_mpi_mesh_transform.generate_corners
unknown file : error : C++ exception with description " [D:\a\conduit\conduit\build\RUN_TESTS.vcxproj]
  file: D:\a\conduit\conduit\src\libs\conduit\conduit_node.cpp
  line: 14021
  message: 
  Invalid child index: 0 (number of children: 0)
  " thrown in the test body.
unknown file : error : C++ exception with description " [D:\a\conduit\conduit\build\RUN_TESTS.vcxproj]
  file: D:\a\conduit\conduit\src\libs\conduit\conduit_node.cpp
  line: 14021
  message: 
  Invalid child index: 0 (number of children: 0)
  " thrown in the test body.
```

File: src/libs/blueprint/conduit_blueprint_mesh_utils.cpp [hunk 1]
```diff
@@ -2066,23 +2066,18 @@ topology::unstructured::points(const Node &n,
...
     const ShapeType topo_shape(ntemp);
 
     std::set<index_t> pidxs;
     if(!topo_shape.is_poly())
     {
         const Node &poffs_node = ntemp["elements/offsets"];
-        temp.set_external(DataType(poffs_node.dtype().id(), 1),
-            (void*)poffs_node.element_ptr(ei));
-        const index_t eoff = temp.to_index_t();
+        const index_t eoff = poffs_node[ei].value();
```
Comment (cyrush) [inline src/libs/blueprint/conduit_blueprint_mesh_utils.cpp]: This is where the issue is, we need the index_t_assessor, not try to ask the node for its child.

**Comment by cyrush** [comment]
@agcapps pushed a fix.

**Comment by cyrush** [comment]
@agcapps  merging this, with the tweaks I made its ready to go.

**Comment by agcapps** [comment]
@cyrush Thanks for the fixes and for merging it in.

**Pull request merged**